openedx/tests/xblock_integration/xblock_testcase.py

```diff
@@ -0,0 +1,202 @@
+        )
+
+        cls.chapter = ItemFactory.create(
+            parent=cls.course,
+            display_name='Overview',
+            category='chapter'
+        )
+
+        cls.urls = {}
+
+        with cls.store.bulk_operations(cls.course.id, emit_signals=False):
+            for section in cls.test_configuration:
+                section_item = ItemFactory.create(
+                    parent=cls.chapter,
+                    display_name=section['urlname'],
+                    category='sequential'
+                )
+
+                cls.urls[section['urlname']] = reverse(
+                    'courseware_section',
+                    kwargs={
+                        'course_id': unicode(cls.course.id),
+                        'chapter': 'Overview',
+                        'section': section['urlname'],
+                    }
+                )
+                print section['urlname']
+                print cls.urls[section['urlname']]
+
+                blocks = section["xblocks"]
+
+                unit_item = ItemFactory.create(
+                    parent=section_item,
+                    display_name="unit_"+section['urlname'],
+                    category='vertical'
+                )
+                for block in blocks:
+                    block_item = ItemFactory.create(
+                        parent=unit_item,
+                        category=block['blocktype'],
+                        display_name=block['urlname']
+                    )
+            cls.course_url = reverse(
+                'courseware_section',
+                kwargs={
+                    'course_id': unicode(cls.course.id),
+                    'chapter': 'Overview',
+                    'section': section['urlname'],
+                }
+            )
+            print cls.course_url
+
+            cls.course_url = reverse(
+                'courseware_section',
+                kwargs={
+                    'course_id': unicode(cls.course.id),
+                    'chapter': u'Overview',
+                    'section': u'two_done_block_test',
+                }
+            )
+            print cls.course_url
+
+    def _containing_section(self, block_urlname):
+        '''
+        For a given block, return the parent section
+        '''
+        for section in self.test_configuration:
+            blocks = section["xblocks"]
+            for block in blocks:
+                if block['urlname'] == block_urlname:
+                    return section['urlname']
+        raise Exception("Block not found "+block_urlname)
+
+    def assertXBlockScreenshot(self, block_urlname, rendering=None):
+        '''
+        As in Bok Choi, but instead of a CSS selector, we pass a
+        block_id. We may want to be able to pass an optional selector
+        for picking a subelement of the block.
+
+        This confirms status code, and that the screenshot is
+        identical.
+
+        TODO: IMPLEMENT
+        '''
+        pass
```

Comment:
Common signature might be nice. I want to be a little bit careful here, though. I'd like the XBlocks test framework to not be edX-platform/bok-choi specific. The goal here is to have a first step towards decoupling XBlock testing from our specific technologies. Very roughly speaking, I'd like to see us define an XBlock test API and build it in edx-platform. Once that's in place, we could start to tease the tests into the XBlock repos themselves, and try to add the same API to xblock-sdk. 

This PR isn't meant to get us all the way there, but at least to start to move us in that direction.

XBlocks and bc have somewhat different requirements. If there is a natural fit for a common signature, I'd do that, but I wouldn't compromise on the design goals of either to achieve that. This is intended as a strawman, and I can already see things wrong with it (`rendering` should be required, and should have a better name). In this case, the compromise I made was: 

* Move from filenames in bc to a string identifier
* Add the ability to specify an XBlock by identifier/url_name, rather than by CSS selector
* Provide the ability to have a CSS subselector
* Keep the names similar enough to show a connection, but have a thin layer which will translate url_name+CSS subselector in the XBlock version to just a CSS selector in the bc version.